### PR TITLE
feat(Docker): Enable Bazel in the Analyzer Docker build

### PR DESCRIPTION
**Pull request by nnobelis**

### CODE REVIEW

File: workers/analyzer/docker/Analyzer.Dockerfile [hunk 1]
```diff
@@ -22,6 +22,7 @@
 # License-Filename: LICENSE
 
 ARG ANDROID_CMD_VERSION=11076708
+ARG BAZEL_VERSION=7.0.1
```
Comment (sschuberth) [inline workers/analyzer/docker/Analyzer.Dockerfile:25]: Why not the latest 7.2.1? If there's a reason, please provide it in the commit message.

File: workers/analyzer/docker/Analyzer.Dockerfile [hunk 2]
```diff
@@ -408,6 +409,25 @@ RUN mkdir -p $DOTNET_HOME/bin \
...
+
+ENV BAZEL_HOME=/opt/bazel
+
+RUN mkdir -p $BAZEL_HOME/bin \
+    && if [ "$(arch)" = "aarch64" ]; then \
+    curl -L https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-arm64 -o $BAZEL_HOME/bin/bazel; \
+    else \
+    curl -L https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-linux-x86_64 -o $BAZEL_HOME/bin/bazel; \
+    fi \
+    && chmod a+x $BAZEL_HOME/bin/bazel
```
Comment (sschuberth) [inline workers/analyzer/docker/Analyzer.Dockerfile:426]: @haikoschol is this still the "right" way to install Bezel, or should maybe Bazelisk be used instead by now?

File: workers/analyzer/docker/Analyzer.Dockerfile (hunk 1 above)
Comment (nnobelis) [reply workers/analyzer/docker/Analyzer.Dockerfile:25]: This is just aligned with ORT :shrug: 

What do you think we should do?

Comment (sschuberth) [reply workers/analyzer/docker/Analyzer.Dockerfile:25]: Hmm, I just checked, and running ORT core tests with Bazel 7.2.1 requires changes to test assets, so I'd propose to stick both to Bazel 7.0.1 for now, but mention the alignment in the commit message.

File: workers/analyzer/docker/Analyzer.Dockerfile (hunk 2 above)
Comment (sschuberth) [reply workers/analyzer/docker/Analyzer.Dockerfile:426]: Let's go ahead with this and eventually improve later, when doing the Bazel 7.2.1 (or later) upgrade.

**Review by sschuberth — APPROVED** [review]

File: workers/analyzer/docker/Analyzer.Dockerfile (hunk 2 above)
Comment (haikoschol) [reply workers/analyzer/docker/Analyzer.Dockerfile:426]: This is [what Bazelisk does](https://github.com/bazelbuild/bazelisk?tab=readme-ov-file#where-does-bazelisk-get-bazel-from) essentially.

I don't think Bazelisk supports installing a specific version. The point of using it is to allow [specifying the Bazel version to run in a repository/workspace](https://github.com/bazelbuild/bazelisk?tab=readme-ov-file#how-does-bazelisk-know-which-bazel-version-to-run).

**Pull request merged**